core/domain/activity_services_test.py

```diff
@@ -0,0 +1,201 @@
+# coding: utf-8
+#
+# Copyright 2014 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from core.domain import activity_domain
+from core.domain import activity_services
+from core.domain import collection_services
+from core.domain import exp_services
+from core.domain import rights_manager
+from core.tests import test_utils
+import feconf
+
+
+class ActivityServicesTests(test_utils.GenericTestBase):
+    """Test the activity services module."""
+
+    EXP_ID_0 = 'EXP_ID_0'
+    EXP_ID_1 = 'EXP_ID_1'
+    COL_ID_2 = 'COL_ID_2'
+
+    def _get_exploration_reference(self, exploration_id):
+        return activity_domain.ActivityReference(
+            feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id)
+
+    def _get_collection_reference(self, collection_id):
+        return activity_domain.ActivityReference(
+            feconf.ACTIVITY_TYPE_COLLECTION, collection_id)
+
+    def _compare_lists(self, reference_list_1, reference_list_2):
+        hashes_1 = [reference.get_hash() for reference in reference_list_1]
+        hashes_2 = [reference.get_hash() for reference in reference_list_2]
+        self.assertEqual(hashes_1, hashes_2)
+
+    def setUp(self):
+        """Publish two explorations and one collection."""
+        super(ActivityServicesTests, self).setUp()
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.signup(self.MODERATOR_EMAIL, self.MODERATOR_USERNAME)
+        self.moderator_id = self.get_user_id_from_email(self.MODERATOR_EMAIL)
+        self.set_moderators([self.MODERATOR_USERNAME])
+
+        self.save_new_valid_exploration(self.EXP_ID_0, self.owner_id)
+        self.save_new_valid_exploration(self.EXP_ID_1, self.owner_id)
+        self.save_new_valid_collection(
+            self.COL_ID_2, self.owner_id, exploration_id=self.EXP_ID_0)
+
+    def test_basic_operations(self):
```

Comment:
I can see this becoming two tests:

``test_update_featured_activity_references_correctly_promotes_activities``
``test_update_featured_activity_references_can_clear_existing_featured_activities``